File: app/src/main/java/org/sil/storyproducer/controller/SplashScreenActivity.java
package org.sil.storyproducer.controller;

import android.content.Intent;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import org.sil.storyproducer.R;
import org.sil.storyproducer.model.Workspace;

public class SplashScreenActivity extends AppCompatActivity {
    //Time in ms for splash screen to be shown
    private static int TIME_OUT = 300;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        Workspace.INSTANCE.initializeWorkspace(this);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                if (!Workspace.INSTANCE.getWorkspace().isDirectory()){
                    Intent intent = new Intent(SplashScreenActivity.this, WorkspaceDialogUpdateActivity.class);
                    startActivity(intent);
                } else if (! Workspace.INSTANCE.getRegistration().getRegistrationComplete()) {
                    Intent intent = new Intent(SplashScreenActivity.this, RegistrationActivity.class);
                    startActivity(intent);
                } else {
                    Intent intent = new Intent(SplashScreenActivity.this, MainActivity.class);
                    startActivity(intent);
                }
            }
        }, TIME_OUT);
    }
}

File: app/src/main/java/org/sil/storyproducer/controller/SplashScreenActivity.kt
package org.sil.storyproducer.controller

import android.content.SharedPreferences
import android.content.pm.PackageManager
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.Spinner
import android.widget.TextView
import androidx.preference.PreferenceManager
import org.sil.storyproducer.R
import org.sil.storyproducer.activities.BaseActivity
import org.sil.storyproducer.model.Workspace
import java.util.Locale

class SplashScreenActivity : BaseActivity() {

    private lateinit var languageSpinner: Spinner
    private lateinit var sharedPreferences: SharedPreferences

    // Map of language names to locale codes
    private val languages = mapOf(
        "" to "",
        "English" to "en",
        "Français (French)" to "fr",
//        "Italiano (Italian)" to "it",
//        "Deutsch (German)" to "de",
        "Español (Spanish)" to "es",
//        "日本語 (Japanese)" to "ja",
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // get the shared preferences used by the Settings activity
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)

        if (Workspace.isInitialized) {
            showMain()
        } else {
            setContentView(R.layout.activity_splash_screen)

            try {
                val title: TextView = findViewById(R.id.version)
                title.text = packageManager.getPackageInfo(packageName, 0).versionName
            } catch (e: PackageManager.NameNotFoundException) {
                e.printStackTrace()
            }

            languageSpinner = findViewById(R.id.language_spinner)

            // get the current language settings
            val currentLocale = Locale.getDefault()
            val languageCode = currentLocale.language // e.g., "en", "es", "fr"
            val countryCode = currentLocale.country   // e.g., "US", "ES", "FR"

            // Set any saved language which takes precedence
            val savedLanguage = sharedPreferences.getString("language", "") ?: ""
            if (savedLanguage != "") {
                LanguageHelper.setLocale(applicationContext, savedLanguage)
                initWorkspace()
                return
            }

            // If system default language is supported then remember and use it
            if (languages.containsValue(languageCode)) {
                LanguageHelper.setLocale(applicationContext, languageCode)
                initWorkspace()
                return
            }

            // User needs to select a language so make the language selection spinner etc. visible
            languageSpinner.visibility = View.VISIBLE
            var languageIcon = findViewById<ImageView>(R.id.language_icon)
            languageIcon.visibility = View.VISIBLE
            var selectLang = findViewById<TextView>(R.id.select_lang)
            selectLang.visibility = View.VISIBLE

            // Populate Spinner
            val adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, languages.keys.toList())
            adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
            languageSpinner.adapter = adapter
//            languageSpinner.prompt = "Select a language"

            // Select empty language string in Spinner
            val currentLanguageIndex = languages.values.indexOf("")
            if (currentLanguageIndex != -1) {
                languageSpinner.setSelection(currentLanguageIndex)
            }

            // Handle language selection
            languageSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
                override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
                    val selectedLanguageCode = languages.values.toList()[position]
                    if (selectedLanguageCode != "") {
                        LanguageHelper.setLocale(applicationContext, selectedLanguageCode)
                        initWorkspace()
                    }
                }

                override fun onNothingSelected(parent: AdapterView<*>?) {}
            }
        }
    }
}
